crates/sui-graphql-rpc/src/extensions/query_limits_checker.rs

```diff
@@ -255,6 +308,49 @@ impl<'a> LimitsTraversal<'a> {
         Ok(())
     }
 
+    /// multiGetObject query can only pass a number of object keys that does not exceed the max
+    /// page size. This checks for the number of keys and deducts the raw size of the keys from
+    /// the payload size.
+    fn check_multiget_objs_arg(
```

Comment:
What we care about is the number of elements in the list that has been input here, which means this function should look something like so:

- If `value` is a variable, look it up to get the literal it points to.
- If that literal is a list, or the value was a list itself, return that list's length.
- Otherwise, return some default size (probably `1`).

I don't know why payload sizes are coming into the equation here.

Reply:
> Let's not make this the same as max_page_size, the reason we factored it out into its own config is that it can/should be different -- we can start by setting this to 500 or something.

> I don't know why payload sizes are coming into the equation here.

Maybe this was the wrong place, but if we're starting with 500 keys allowed and we're not deducting them from the query payload, then any query with more than 60-70 keys (in a pretty formatted query) will fail to pass the query payload limit check.

Reply:
Recording our offline conversation: These are two orthogonal concerns -- what this limit is protected us against (large numbers of output nodes), and how the specific values we've chosen for the limit interact with other unrelated limits (like query payload size).

In this case, if the numbers we've chosen for these limits mean there's a large dead region in the state space (states where only one of the limits is valid, but not both), then we should tweak what the limits are, not change what we're protecting against.